database/budget.go

```diff
@@ -2,32 +2,42 @@ package database
 
 import (
 	"database/sql"
+	"errors"
 	"time"
 )
 
 type BudgetInterval struct {
 	StartDate time.Time
 	EndDate   time.Time
+	Name      string
+	EntityId  Id
 }
 
-func (database *Database) QueryCurrentBudgetInterval() (*BudgetInterval, error) {
-	row := database.QueryRow("SELECT * FROM autobudget ORDER BY startDate DESC LIMIT 1")
+func (database *Database) QueryCurrentBudgetInterval(name string, entityId Id) (*BudgetInterval, error) {
```

Comment:
Why not use the `swapperType` as parameter of the function for type safety and cast to string in the query?

Reply:
There are cyclic dependency issues and I'm not sure about moving the `SwapperType` into the db package